src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -762,32 +759,32 @@ InferenceProfiler::Measure(
 
   TimestampVector current_timestamps;
   RETURN_IF_ERROR(manager_->SwapTimestamps(current_timestamps));
+  all_timestamps_.insert(
+      all_timestamps_.end(), current_timestamps.begin(),
+      current_timestamps.end());
 
   RETURN_IF_ERROR(Summarize(
-      current_timestamps, start_status, end_status, start_stat, end_stat,
-      status_summary, measurement_window_ms));
+      start_status, end_status, start_stat, end_stat, status_summary,
+      window_start_ns.count(), window_end_ns.count()));
 
   return cb::Error::Success;
 }
 
 cb::Error
 InferenceProfiler::Summarize(
-    const TimestampVector& timestamps,
     const std::map<cb::ModelIdentifier, cb::ModelStatistics>& start_status,
     const std::map<cb::ModelIdentifier, cb::ModelStatistics>& end_status,
     const cb::InferStat& start_stat, const cb::InferStat& end_stat,
-    PerfStatus& summary, long long measurement_window_ms)
+    PerfStatus& summary, uint64_t window_start_ns, uint64_t window_end_ns)
 {
   size_t valid_sequence_count = 0;
   size_t delayed_request_count = 0;
 
   // Get measurement from requests that fall within the time interval
-  std::pair<uint64_t, uint64_t> valid_range;
-  MeasurementTimestamp(timestamps, &valid_range, measurement_window_ms);
+  std::pair<uint64_t, uint64_t> valid_range{window_start_ns, window_end_ns};
```

Comment:
This is a big change that basically says the window start and stop as measured above will be used as the window boundaries. All requests that end in between are considered valid/included in window and throughput metric calculations.